show/main.py

```diff
@@ -1227,10 +1227,19 @@ def users(verbose):
 
 @cli.command()
 @click.option('--since', required=False, help="Collect logs and core files since given date")
+@click.option('--force', is_flag=True, help="Enable hardware register dump even with system interruption")
+@click.option('--yes', is_flag=True, help="Enable force generating dump option")
 @click.option('--verbose', is_flag=True, help="Enable verbose output")
-def techsupport(since, verbose):
+def techsupport(since, verbose, force, yes):
     """Gather information for troubleshooting"""
     cmd = "sudo generate_dump -v"
+    if force:
+        if yes:
+            cmd += " -f"
+        else:
+            if click.confirm('Proceed with any system interruption?', abort=True):
```

Comment:
With `abort=True`, this does not need to be an `if` statement. If the user declines, the command will abort. If you want to print a message and exit, you can remove `abort=True` and add an `else` statement.

Reply:
Aborting when the user declines, is ok but we can add a message before exit for this case.